src/object/pushbutton.cpp

```diff
@@ -51,8 +55,12 @@ PushButton::get_settings()
   ObjectSettings result = MovingSprite::get_settings();
 
   result.add_script(_("Script"), &script, "script");
+  result.add_bool(_("Upside down"), &m_upside_down, "upside-down");
+
+  result.reorder({"script", "upside-down", "x", "y"});
 
-  result.reorder({"script", "x", "y"});
+  if (m_upside_down && m_flip == NO_FLIP)
+    FlipLevelTransformer::transform_flip(m_flip);
```

Comment:
This code can be moved to `after_editor_set` method. Also another thing, apparently pushbutton remains flipped even if you uncheck the Upside down option. This can be fixed by changing `if (m_upside_down && m_flip == NO_FLIP)` to `if ((m_upside_down && m_flip == NO_FLIP) || (!m_upside_down && m_flip == VERTICAL_FLIP))`.